django/db/migrations/autodetector.py

```diff
@@ -674,6 +683,16 @@ def generate_created_models(self):
                         if field.primary_key:
                             primary_key_rel = field.remote_field.model
                         elif not field.remote_field.parent_link:
+                            attname = None
+                            if hasattr(field, "get_attname"):
+                                attname = field.get_attname()
+                            if field_name in referenced_base_fields or (
+                                attname is not None
+                                and attname in referenced_base_fields
+                            ):
+                                # Keep this field inline in CreateModel so the
+                                # GeneratedField's expression can see it.
+                                continue
```

Comment:
Hi, thanks for the ticket & patch, however …

I don't think this approach will be accepted. FKs are required to be deferred to avoid circular references with the model state.

I believe we may need to just defer the creation of the generated field in the same way as FKs are. We may be able to get away with having all generated fields, that refer to a FK, ordered **last**.